DEVELOPMENT.md

```diff
@@ -0,0 +1,536 @@
+./runtests.sh format
+
+# Fix formatting
+./runtests.sh format --fix
+
+# Run linting
+./runtests.sh lint
+
+# Fix linting issues
+./runtests.sh lint --fix
+```
+
+### 5. Commit Changes
+
+```bash
+git add .
+git commit -m "Description of changes"
+```
+
+### 6. Push and Create PR
+
+```bash
+git push origin feature-name
+```
+
+Then open a pull request on GitHub against the `main` branch.
+
+## Project Structure
+
+```
+nyrkio/
+├── backend/                  # FastAPI backend application
+│   ├── api/                 # API endpoints
+│   │   ├── api.py          # Main FastAPI app
+│   │   ├── admin.py        # Admin endpoints
+│   │   ├── billing.py      # Billing endpoints
+│   │   ├── changes.py      # Change detection endpoints
+│   │   └── ...
+│   ├── auth/                # Authentication logic
+│   ├── core/                # Core business logic
+│   ├── db/                  # Database models and operations
+│   ├── github/              # GitHub integration
+│   ├── notifiers/           # Notification services (Slack, GitHub)
+│   ├── tests/               # Unit tests
+│   ├── integration_tests/   # Integration tests
+│   ├── hunter/              # Hunter submodule (change detection)
+│   ├── pyproject.toml       # Poetry dependencies
+│   ├── runtests.sh          # Test runner script
+│   └── entrypoint.sh        # Docker entrypoint
+├── frontend/                 # React frontend application
+│   ├── src/                 # React source code
+│   ├── public/              # Static assets
+│   ├── package.json         # npm dependencies
+│   ├── vite.config.js       # Production Vite config (proxies to nyrk.io)
+│   └── vite.config.test.js  # Local dev Vite config (proxies to localhost:8001)
+├── nginx/                    # Nginx configuration
+├── etc/                      # Management scripts
+│   ├── nyrkio_backend.py    # Backend server manager (standalone)
+│   ├── nyrkio_frontend.py   # Frontend server manager (with deployment modes)
+│   ├── nyrkio_docker.py     # Docker stack manager (full stack)
+│   └── test_management_scripts.py  # Test suite for management scripts
+├── docker-compose.dev.yml    # Docker Compose for development
+├── .env.backend              # Environment configuration (create this)
+├── .backend.pid              # Backend process ID (auto-generated)
+└── README.md                 # Main documentation
+```
+
+## Configuration
+
+### Environment Variables
+
+Configuration is managed through the `.env.backend` file:
+
+```bash
+# Database Configuration
+DB_URL=mongodb://mongodb.nyrkio.local:27017/mongodb
+DB_NAME=nyrkiodb
+
+# API Keys
+POSTMARK_API_KEY=           # Email service
+GITHUB_CLIENT_SECRET=       # GitHub OAuth
+SECRET_KEY=                 # JWT signing key
+
+# Optional
+HUNTER_CONFIG=              # Hunter configuration
+GRAFANA_USER=               # Grafana integration
+GRAFANA_PASSWORD=           # Grafana password
+
+# Development Settings
+API_PORT=8001               # Backend server port
+```
```

Comment:
One problem here is the webhooks and other callbackss from almost any of these services. If you run this on localhost, click on the link to login via GH, then GH will send you back to a hard coded nyrkio.com/login/success/or/something... and you will be logged in on the prod server but certainly not localhost.

So what we really need is a separate app, app id, everything ... and this nyrkiö-test-app has its own stripe, database, domain name, everything.